databricks/koalas/spark.py

```diff
@@ -677,6 +678,69 @@ def explain(self, extended: Optional[bool] = None, mode: Optional[str] = None):
         else:
             self._kdf._internal.to_internal_spark_frame.explain(extended, mode)
 
+    def apply(self, func, index_col=None):
+        """
+        Applies a function that takes and returns a Spark DataFrame. It allows natively
+        apply a Spark function and column APIs with the Spark column internally used
+        in Series or Index.
+
+        .. note:: set `index_col` and keep the column named as so in the output Spark
+            DataFrame to avoid using the default index to prevent performance penalty.
+            If you omit `index_col`, it will use default index which is potentially
+            expensive in general.
+
+        .. note:: it will lose column labels. This is a synonym of
+            ``func(kdf.to_spark(index_col)).to_koalas(index_col)``.
+
+        Parameters
+        ----------
+        func : function
+            Function to apply the function against the data by using Spark DataFrame.
+
+        Returns
+        -------
+        DataFrame
```

Comment:
Just my opinion, how about `Koalas DataFrame` or `ks.DataFrame` rather just `DataFrame` because we're describing this functions as "Applies a function that takes and returns a `Spark DataFrame`" ??
I think maybe It can be confused whether the return type is `Spark DataFrame` or `Koalas DataFrame`.